include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp

```diff
@@ -191,11 +191,11 @@ class aligned_sequence_builder
             std::pair<size_t, size_t>{trace_it.coordinate()};
 
         assign_unaligned(std::get<0>(res.alignment),
-                         fst_rng | views::slice(res.first_sequence_slice_positions.first,
-                                                res.first_sequence_slice_positions.second));
+                         std::views::all(fst_rng) | views::slice(res.first_sequence_slice_positions.first,
```

Comment:
Not type reduce here?

Reply:
This one is funny, because `fst_rng` is already of type `type_reduce<urange>`

Reply:
I need to use `std::views::all(fst_rng)`, because otherwise `fst_rng` in `fst_rng | views::slice` would be a lvalue, which would when not be a view but a viewable_range. This means we should ALWAYS call `std::views::all`, before piping it into something.

Reply:
> This semantic difference can cause different resulting views

That sounds broken!

Reply:
I think this was one of the reasons why eric didn't want to allow lvalues, because the whole concept of views is that they are rvalues. If you pass a `view &` into something it will not be a view, but a `range &` that must be made into a view by `std::views::all`.